cardano-crypto-class/src/Cardano/Crypto/DSIGN/SchnorrSecp256k1.hs

```diff
@@ -144,25 +141,23 @@ instance DSIGNAlgorithm SchnorrSecp256k1DSIGN where
         pure . SigSchnorrSecp256k1 $ sigPSB
   {-# NOINLINE verifyDSIGN #-}
   verifyDSIGN () (VerKeySchnorrSecp256k1 pubkeyPSB) msg (SigSchnorrSecp256k1 sigPSB) =
-    unsafeDupablePerformIO . psbUseAsCPtr pubkeyPSB $ \pkp -> 
-      psbUseAsCPtr sigPSB $ \sigp -> do
+    unsafeDupablePerformIO . psbUseAsSizedPtr pubkeyPSB $ \pkp ->
+      psbUseAsSizedPtr sigPSB $ \sigp -> do
         let bs = getSignableRepresentation msg
-        let sigp' :: Ptr CUChar = castPtr sigp
         res <- useAsCStringLen bs $ \(msgp, msgLen) -> do
           pure $ secpSchnorrSigVerify ctxPtr 
-                                      sigp' 
+                                      sigp
                                       (castPtr msgp) 
                                       (fromIntegral msgLen)
-                                      (castPtr pkp)
+                                      pkp
         pure $ if res == 0
           then Left "Schnorr signature failed to verify."
           else pure ()
   {-# NOINLINE genKeyDSIGN #-}
   genKeyDSIGN seed = runMonadRandomWithSeed seed $ do
```

Comment:
IIUC, we aren't using anything from `MonadRandom` here anymore, so `runMonadRandomWithSeed` is redundant, isn't it?